xbmc/filesystem/File.cpp

```diff
@@ -283,7 +288,10 @@ bool CFile::Open(const CURL& file, const unsigned int flags)
         return m_pFile->Open(url);
       }
     }
-    m_pFile = CFileFactory::CreateLoader(url);
+    if (m_pFile)
+      m_pFile->Close();
+    else
+      m_pFile = CFileFactory::CreateLoader(url);
```

Comment:
This only concerns CurlFile, right? IMO it should be in that class, not here.

And if, for some reason, it really needs to be here please add a proper comment. In one month no one will know why it is the way it is.